### ISSUE TITLE
Use hyphen-minus within ISO 8601 dates

### ISSUE THREAD

**Issue by jesus2099** [kellnerd/musicbrainz-scripts#38]
I think it's maybe more convenient and less subject to conflict to keep hyphen-minus within ISO 8601 dates (YYYY-MM-DD) and stop replacing them with _true_ hyphen.

https://community.metabrainz.org/t/correct-hyphen-unicode-hyphen-or-hyphen-minus/19610/68?u=jesus2099
https://tickets.metabrainz.org/browse/MBS-12152

**Comment by jesus2099** [comment]
At least it's worth discussing.

**Comment by kellnerd** [comment]
I am happy to change my script to leave dates alone (although that would be a bit annoying to implement with my current approach) if that is the outcome of a community discussion.
But I don't want to simply change this after 3 years of heavy usage and leave people confused (whether the script stopped working or if they should revert all their previous edits) without a clear consensus.

**Comment by jesus2099** [comment]
Yes, we sure have to wait! 😁👍

**Comment by raelthelamb** [comment]
(Sorry for the long comment.)

### First some factual stuff:

The only relevant thing I can find in the [Unicode Standard](https://www.unicode.org/versions/Unicode15.0.0/ch06.pdf) (Section 6.2, hyphens and dashes) is
> When typesetting text, U+2010 hyphen is preferred over U+002D hyphen-minus.

which I guess we all know. I don't find anything more specific referring to dates.

[ISO 8601](https://web.archive.org/web/20171020084445/https://www.loc.gov/standards/datetime/ISO_DIS%208601-1.pdf) regarding date formats has in Section 3.4.1

> The representations specified in this International Standard make use of graphic characters as specified in 3.4. Note that, except for “hyphen”, “minus” and “plus-minus”, these characters are part of the ISO/IEC 646 repertoire.
> In an environment where use is made of a character repertoire based on ISO/IEC 646, “hyphen” and “minus” are both mapped onto “hyphen-minus”. Representations with a “plus-minus” shall only be used in such environment if the interchange repertoire includes “plus-minus”.

and

> NOTE 2 Encoding of characters for the interchange of dates and times is not in the scope of this International Standard

(The link and the quotes refer to a draft, because the final version is, incredibly, paywalled.) The meaning of the paragraph in the first quote is disputed (e.g. [here ](https://stackoverflow.com/questions/77284818/should-hypen-minus-u002d-or-hypen-u2010-be-used-for-iso-8601-datetimes )and [here](https://tickets.metabrainz.org/browse/MBS-12152)), in particular what it means for a character repertoire to be "based on ISO/IEC 646", and whether the large Unicode character set that MB uses is based on this or not. In the second quote, I am not sure if "interchange" includes display / typesetting.

### And secondly an opinion:

When talking about display / typesetting (and not e.g. filenames), I can't really think of a reason to treat the hyphen in a date differently from other hyphens. So given that the decision to use U+2010 for hyphens in words seems fixed, I think it should be used in dates too.

(Were it completely up to me though, I would consistently use U+002D for all hyphens. Even the typesetting software TeX will actually replace U+2010 in input by U+002D in pdf output ... the Unicode standard itself looks like a TeX-produced document to me, which would explain why it always uses U+002D.)

Probably goes without saying, but if it is decided to use U+002D in dates, it would be lovely if the script could be updated to actually change the U+2010s back!

**Comment by jesus2099** [comment]
> The only relevant thing I can find in the [Unicode Standard](https://www.unicode.org/versions/Unicode15.0.0/ch06.pdf) (Section 6.2, hyphens and dashes) is
> > When typesetting text, U+2010 hyphen is preferred over U+002D hyphen-minus.
> which I guess we all know. I don't find anything more specific referring to dates.

Yes but this is really just about language in general (they also talk about writing direction, CJK punctuation, etc.), not about dates.

I think we should at least align with MBS, that has always displayed dates with U+002D HYPHEN-MINUS, at the moment: https://tickets.metabrainz.org/browse/MBS-12152
